doc/tutorial/conv_arithmetic.txt

```diff
@@ -0,0 +1,597 @@
+* padding size :math:`p`,
+
+is obtained by applying both the rule we derived for strides and the one we
+derived for the padding. In other words,
+
+.. math::
+
+    o = \left\lfloor \frac{i + 2p - k}{s} \right\rfloor + 1.
+
+Here is what it looks like for :math:`i = 5`, :math:`k = 3`, :math:`s = 2` and
+:math:`p = 1`:
+
+.. figure:: conv_arithmetic_figures/padding_strides.gif
+
+Sometimes numbers are mean
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You may have noticed that the floor function in the general case formula will
+produce the same output shape for multiple input shapes. More specifically, if
+:math:`i + 2p - k` is a multiple of :math:`s`, then any input size
+:math:`j = i + a, \quad a \in \{0,\ldots,s - 1\}` will produce the same output
+size. Note that this ambiguity applies only for :math:`s > 1`.
+
+As an example, here is what a convolution looks like for :math:`i = 6`,
+:math:`k = 3`, :math:`s = 2` and :math:`p = 1`:
+
+.. figure:: conv_arithmetic_figures/padding_strides_odd.gif
+
+Despite having different input sizes, this convolution and the previous
+convolution share the same output size.
+
+This may seem like a curiosity, but keep that in mind, as it will become
+important when we examine *transposed convolutions*.
+
+Transposed convolution: convolution turned on its head
+======================================================
+
+Every convolution boils down to an efficient implementation of a matrix
+operation. Take for instance the very first convolution presented in this
+tutorial. If the input and output were to be unrolled into vectors from left to
+right, top to bottom, the matrix representation of the convolution would look
+like
+
+.. math::
+
+    \begin{pmatrix}
+    w_{0,0} & 0       & 0       & 0       \\
+    w_{0,1} & w_{0,0} & 0       & 0       \\
+    w_{0,2} & w_{0,1} & 0       & 0       \\
+    0       & w_{0,2} & 0       & 0       \\
+    w_{1,0} & 0       & w_{0,0} & 0       \\
+    w_{1,1} & w_{1,0} & w_{0,1} & w_{0,0} \\
+    w_{1,2} & w_{1,1} & w_{0,2} & w_{0,1} \\
+    0       & w_{1,2} & 0       & w_{0,2} \\
+    w_{2,0} & 0       & w_{1,0} & 0       \\
+    w_{2,1} & w_{2,0} & w_{1,1} & w_{1,0} \\
+    w_{2,2} & w_{2,1} & w_{1,2} & w_{1,1} \\
+    0       & w_{2,2} & 0       & w_{1,2} \\
+    0       & 0       & w_{2,0} & 0       \\
+    0       & 0       & w_{2,1} & w_{2,0} \\
+    0       & 0       & w_{2,2} & w_{2,1} \\
+    0       & 0       & 0       & w_{2,2} \\
+    \end{pmatrix}^T
+
+where :math:`w_{i,j}` is the element of the kernel located at row :math:`i` and
+column :math:`j`. (*Note: the matrix has been transposed for formatting
+purposes.*) This linear operation takes a 16-dimensional vector as input and
+produces a 4-dimensional vector as output.
+
+What would happen if we were to transpose this matrix and feed the corresponding
+linear operation a 4-dimensional vector as input? We would get a 16-dimensional
+vector as output. This is what we call a **transposed convolution**.
```

Comment:
Consider showing the corresponding matrix product, so that the reader can refer to it in the following section and verify that the operation shown in the figure is the same as the one exposed here.